### PR TITLE
Validation and Specs

### PR DESCRIPTION
**Pull request by GeekyHacks**
In the following PR I did the following:
- SetUp the project for Rspec testing running `bundler i` then `bin/rails g rspec:install` after adding the Rspec gem `gem 'rspec-rails'` to the gemfile
- Added the spec test cases then adding the validations to the User and Post models
- Tested all test cases with `rspec` and got all passed
- Fixed linter errors

### CODE REVIEW

File: app/models/like.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  after_save :update_comments_counter
```
Comment (ademibowale) [inline app/models/like.rb]: - [ ] Great job on your project ,you have done a great job so far but there are still some isues you still need to address ,please kindly add Rspect unit test that describe the like ,as you can see  acccording to the project requirement its mandatory that  Any custom methods in the models should have a corresponding unit test.

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+
+  after_save :update_comments_counter
```
Comment (ademibowale) [inline app/models/comment.rb]: - [ ] Great job on your project, you have done a great job so far, but there are still some issues you still need to address, please kindly add the Rspect unit test that describes the Comment, as you can see according to the project requirement it is mandatory that  Any custom methods in the models should have a corresponding unit test.kindly add unit specs for all of your models' methods consequently each model should have a corresponding unit test

**Review by ademibowale — CHANGES_REQUESTED** [review]
## STATUS: Changes Required :recycle:

### Hi @GeekyHacks  , 👋 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

image


### To Highlight 🥇

✅ Descriptive Pull Request title and summary. ✔️
✅ Good use of Github flow ✔️
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.

### Changes Required

- [x]  _Please check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

⚠️⚠️ Important Note ⚠️⚠️
Please don't open the new PR, please make the above changes in this branch and submit this same PR https://github.com/NalianAchei2020/Hello_Microverse/pull/1 for getting another review.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on( @Adebowale ) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/likes_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+  #   context '#update_likes_counter' do
+  #     before :each do
+  #       8.times { Like.create(user: @user, post: @post) }
+  #     end
+
+  #     it 'keeps track of likes and equals 8' do
+  #       expect(@post.likes_counter).to eq 8
+  #     end
+  #   end
```
Comment (cynthiainga) [inline spec/likes_spec.rb:32]: - [OPTIONAL] Kindly delete the commented line of code to keep your code clean and professional 😁 😁

**Review by cynthiainga — APPROVED** [review]
Hi @GeekyHacks, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

- [ ] All tests are passing ✔️ 
- [ ] Correct Gitflow ✔️ 
- [ ] No linters errors ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**